src/app/src/components/Layers/ReferenceImageLayer.js

```diff
@@ -0,0 +1,83 @@
+import { useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import L from './L.DistortableImage.Edit.fix';
+import ReferenceImage from '../../img/raleigh_sanborn_map.jpg';
+
+import { useMapLayer } from '../../hooks';
+import { customizePrototypeIcon } from '../../utils';
+import { updateReferenceImage } from '../../store/mapSlice';
+
+customizePrototypeIcon(L.DistortHandle.prototype, 'ref-handle');
+customizePrototypeIcon(L.DragHandle.prototype, 'ref-handle');
+customizePrototypeIcon(L.FreeRotateHandle.prototype, 'ref-handle');
+customizePrototypeIcon(L.RotateHandle.prototype, 'ref-handle');
+customizePrototypeIcon(L.ScaleHandle.prototype, 'ref-handle');
+
+const convertCornerToStateFormat = corner => [corner.lat, corner.lng];
+const convertCornerFromStateFormat = corner => ({
+    lat: corner[0],
+    lng: corner[1],
+});
+
+export default function ReferenceImageLayerVisbilityWrapper() {
+    const showLayer = useSelector(state => state.map.referenceImage.visible);
+
+    return showLayer ? <ReferenceImageLayer /> : null;
+}
+
+function ReferenceImageLayer() {
+    const dispatch = useDispatch();
+
+    // TODO: Find a way to initialize the image with transparent/outlined enabled
+    const { corners, mode, transparent, outlined } = useSelector(
```

Comment:
Lint errors may be fixed with:

```suggestion
    const { corners, mode } = useSelector(
```